src/Native/src/kernels/stackvm/optimized/riscv64/binary.cpp

```diff
@@ -170,317 +170,341 @@ struct binary_op_div_rvv {
+            binary_impl_fv_f32<Top>(input_a[0], input_b, output, len_b);
         } else {
-            // a is tensor, b is vector
-            assert(in_a_shape.back() == in_b_shape[0]);
-            size_t nch = count / in_b_shape[0];
-            for (size_t w = 0; w < nch; w++) {
-                int n = in_b_shape[0];
-                const float *input_b_tmp = input_b;
-                while (n > 0) {
-                    vl = vsetvl_e32m8(n);
-                    auto v_a = vle32_v_f32m8(input_a, vl);
-                    auto v_b = vle32_v_f32m8(input_b_tmp, vl);
-                    auto v_out = op(v_a, v_b, vl);
-                    vse32_v_f32m8(out, v_out, vl);
-
-                    input_a += vl;
-                    input_b_tmp += vl;
-                    out += vl;
-                    n -= vl;
-                }
+            int loop_n = len_b / len_a;
+            for (int i = 0; i < loop_n; ++i) {
+                binary_impl_vv_f32<Top>(input_a, input_b, output, len_a);
+                input_b += len_a;
+                output += len_a;
             }
         }
     }
-    return ok();
+
+    return 0;
 }
 
 // int32_t
 template <typename Top>
-result<void> optimized_binary_impl(const int32_t *input_a,
-                                   const int32_t *input_b, int32_t *output,
-                                   gsl::span<const size_t> in_a_shape,
-                                   gsl::span<const size_t> in_b_shape,
-                                   gsl::span<const size_t> out_shape) noexcept {
-    Top op;
-    const int count = compute_size(out_shape);
-    size_t vl = 0;
-    auto out = output;
-
-    if (in_a_shape.size() == in_b_shape.size()) {
-        int n = count;
-
-        // both a and b are tensors
-        while (n > 0) {
-            vl = vsetvl_e32m8(n);
-            auto v_a = vle32_v_i32m8(input_a, vl);
-            auto v_b = vle32_v_i32m8(input_b, vl);
-            auto v_out = op(v_a, v_b, vl);
-            vse32_v_i32m8(out, v_out, vl);
-
-            input_a += vl;
-            input_b += vl;
-            out += vl;
-            n -= vl;
-        }
-    } else if (in_a_shape.size() < in_b_shape.size()) {
-        if (is_scalar(in_a_shape)) {
-            //        if (in_a_shape[0] == 1) {
-            // a is scalar, b is tensor
-            int n = count;
-            while (n > 0) {
-                vl = vsetvl_e32m8(n);
-                auto v_b = vle32_v_i32m8(input_b, vl);
-                auto v_out = op(input_a[0], v_b, vl);
-                vse32_v_i32m8(out, v_out, vl);
-
-                input_b += vl;
-                out += vl;
-                n -= vl;
-            }
+int optimized_binary_impl(const int32_t *input_a, const int32_t *input_b,
+                          int32_t *output, gsl::span<const size_t> &in_a_shape,
+                          gsl::span<const size_t> &in_b_shape,
+                          gsl::span<const size_t> &out_shape) noexcept {
+    (void)out_shape;
```

Comment:
Use `[[maybe_unused]]` instead of `(void)`

Reply:
Done